roles/kubernetes/node/tasks/main.yml

```diff
@@ -29,12 +29,54 @@
     state: directory
   when: dynamic_kubelet_configuration
 
-- name: Write kubelet config file (kubeadm)
+- name: gets the kubeadm version
+  command: "{{ bin_dir }}/kubeadm version -o short"
+  register: kubeadm_output
+
+- name: sets kubeadm api version to v1beta1
+  set_fact:
+    kubeadmConfig_api_version: v1beta1
+  when: kubeadm_output.stdout is version('v1.13.0', '>=')
+  tags:
+    - kubelet
+    - kubeadm
+
+- name: Write kubelet environment config file (kubeadm)
   template:
-    src: kubelet.kubeadm.env.j2
+    src: "kubelet.kubeadm.env.{{ kubeadmConfig_api_version }}.j2"
     dest: "{{ kube_config_dir }}/kubelet.env"
     backup: yes
   notify: restart kubelet
+  when: kubeadm_output.stdout is version('v1.13.0', '>=')
+  tags:
+    - kubelet
+    - kubeadm
+
+- name: Write kubelet environment config file (kubeadm)
+  template:
+    src: "kubelet.kubeadm.env.j2"
+    dest: "{{ kube_config_dir }}/kubelet.env"
+    backup: yes
+  notify: restart kubelet
+  when: kubeadm_output.stdout is version('v1.13.0', '<')
+  tags:
+    - kubelet
+    - kubeadm
+
+- name: Write kubelet init config file for kubeadm
+  template:
+    src: "kubelet-kubeadm-config.{{ kubeadmConfig_api_version }}.yaml.j2"
+    dest: "{{ kube_kubeadm_temp_config_dir }}/kubelet-kubeadm-config.yaml"
+  when: kubeadm_output.stdout is version('v1.13.0', '>=')
+  tags:
+    - kubelet
+    - kubeadm
+
+- name: kubeadm | Merge all configs for kubeadm in one file
```

Comment:
why add this on work node?

Reply:
this role runs not only on worker node, this role runs on all nodes, but all ansible variables that needed for kubelet and kubelet itself are defined on this role, i do not wont to make huge changes to all structure of kubespray just to add this functionality, but if you think there is a better way i open for new solution )